resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -38,5 +42,34 @@ object DbReader {
         .query[Runtime]
         .stream
         .transact(xa)
+
+    override def getBucketsToDelete: Stream[F, BucketToRemove] =
+      sql"""select googleProject, stagingBucket from CLUSTER WHERE status="Deleted" and destroyedDate < now() - interval 15 DAY;"""
+        .query[BucketToRemove]
+        .stream
+        .transact(xa)
+
+    // Return all clusters that has no non "DELETED" | `ERROR` nodepool
+    override def getK8sClustersToDelete: Stream[F, KubernetesClusterId] =
+      sql"""SELECT kc.id, kc.location, kc.clusterName
```

Comment:
We might need to iterate on this query.. I had wanted to add a configurable "grace period" during which we leave empty clusters running.

Also I think all clusters will have at least 1 nodepool at all times, so this query may never trigger any clusters.

Reply:
so should the criteria be more than one nodepool?

Reply:
maybe something like:
```
select kc.id, kc.location, kc.clusterName
from KUBERNETES_CLUSTER kc
where (
  select count(*) from NODEPOOL np WHERE np.clusterId = kc.id AND np.status <> 'Error' AND np.status <> 'Deleted'
) = 1
AND kc.status <> 'Error' AND kc.status <> 'Deleted'
```

Just thinking out loud, maybe there is a better way to distinguish the dummy nodepool than just count.

Reply:
I believe that the `NODEPOOL` table has an `isDefault` field, which may help.

Reply:
@rtitle what is `kc.status <> 'Error' AND kc.status <> 'Deleted'` for? I thought we're deleting the cluster regardless of its current status as long as it's not being actively used anymore